velox/vector/ComplexVector.h

```diff
@@ -249,8 +249,8 @@ class RowVector : public BaseVector {
   /// Push all dictionary encoding to the leave vectors of a RowVector tree
   /// (i.e. we traverse the tree consists of RowVectors, possibly wrapped in
   /// DictionaryVector, and no traverse into other complex types like array or
-  /// map children).  If the wrapper introduce nulls on RowVector, we don't push
-  /// the dictionary into that RowVector.  The input vector should not contain
+  /// map children).  When new nulls are introduced on RowVector, we combine it
+  /// with the existing nulls on RowVector.  The input vector should not contain
```

Comment:
@Yuhta Would you document the requirements for the 'input' vector. I assumed it should be a RowVector possibly wrapped in dictionaries and consts, but looks like it can be any vector. Is this so?

Reply:
Yes it can be any vector, in that case we combine the adjacent DICT layers together.  So basically we combining these 2:
1. DICT(DICT) -> DICT
2. DICT(ROW(DICT)) -> DICT

Reply:
I see. Let's update the documentation to make this clear.

> Push all dictionary encoding to the leaf vectors of a RowVector tree

This suggests that input is a RowVector, but apparently that's not necessarily the case. 

Do we have test coverage for non-ROW vectors?

Reply:
Yes the first 2 cases in the unit test is for that